Add bookings

numerouno-life opened the pull request.

numerouno-life committed:
создал БД со всеми нужными сущностями. Создал сущности Booking и Comment ко всем сущностям сделал аннотации

numerouno-life committed:
Выполнил ТЗ проекта №15

irek-valiev commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+        log.info("Добавление нового бронирования");
+        User user = findUserById(userId);
+        Item item = findItemById(bookingDtoIn.getItemId());
+        if (Boolean.FALSE.equals(item.getAvailable())) {
```
item.getAvailable() сам возвращает булево значение, его не нужно сравнивать с Boolean.FALSE. Нужно инвертировать его значение

irek-valiev commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+        if (Boolean.FALSE.equals(item.getAvailable())) {
+            throw new ValidationException("Вещь не доступна для бронированя");
+        }
+        if (bookingDtoIn.getStart() == bookingDtoIn.getEnd()) {
```
эту валидацию сделай самой первой в методе, чтобы зря по хранилищам не ходить за пользователем и вещью

irek-valiev commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    private User findUserById(Long userId) {
+        return UserMapper.toUser(userService.getUserById(userId));
```
может инжектнуть сюда userRepository, а не userService?? Какой-то лишний маппинг model в DTO и обратно получается

irek-valiev commented on src/main/java/ru/practicum/shareit/item/service/ItemServiceImpl.java:
```diff
@@ -2,57 +2,180 @@
+            return CommentMapper.toCommentDtoOut(commentRepository.save(
+                    CommentMapper.toComment(commentDtoIn, booking.getItem(), booking.getBooker())));
+        } catch (Exception e) {
+            throw new ValidationException("Бронь не найдена");
```
добавь в сообщение какую бронь не нашли

numerouno-life committed:
Убрал лишний маппинг при поиске User
Упростил логику проверки доступности в addNewBooking и перенес очередность валидации
